### ISSUE TITLE
Executor queue grows unlimited and cause OoM

### ISSUE THREAD

**Issue by tomasz-sadura** [awslabs/dynamodb-import-export-tool#2]
ExecutorCompletionService in DynamoDBConsumer have tasks submitted, but the results are never taken from the executor completion queue. This may cause OutOfMemory errors for large tables.

**Comment by jseed** [comment]
+1 For this issue.
Attempting to copy larger tables causes the import export tool to fail, throwing `java.lang.OutOfMemoryError: GC overhead limit exceeded`

**Comment by viyatgandhi** [comment]
@JSeed can you tell me how big of a table can cause this problem ? I am thinking to use this in prod for some replication issue, although it worked fine in our test env on small table.

**Comment by amcp** [comment]
To limit the size of your executor queue and avoid OutOfMemory errors, you can take the following steps:

1. Increase the memory you allocate to the JVM
2. In case you have already maxed out JVM allocation, try doing the table copy in multiple sections. Tune the number of sections until the memory footprint of each section fits within your JVM allocation.
3. You can copy sections serially on the same host, or in parallel on multiple hosts.

**Comment by amcp** [comment]
I agree that this library should use numWriteThreads somehow to limit the queue length.
